# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1222794666 14400
#      Tue Sep 30 13:11:06 2008 -0400
# Node ID 75014ec4ac84d4d4637b5423fb6e661726553b33
# Parent  0d37c99fc06faa560de017f367ff4ffa92b97a2f
__magick_read__.cc: only define magick_to_octave_value functions if HAVE_MAGICK is defined

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-09-30  John W. Eaton  <jwe@octave.org>
+
+	* DLD-FUNCTIONS/__magick_read__.cc: Only define
+	magick_to_octave_value if HAVE_MAGICK is defined.
+
 2008-09-29  John W. Eaton  <jwe@octave.org>
 
 	* symtab.cc (symbol_table::fcn_info::fcn_info_rep::load_class_method):
 	Call load_class_constructor if name and dispatch_type are the same.
 
 2008-09-29  David Bateman  <dbateman@free.fr>
 
 	* symtab.cc (octave_value symbol_table::find_function 
diff --git a/src/DLD-FUNCTIONS/__magick_read__.cc b/src/DLD-FUNCTIONS/__magick_read__.cc
--- a/src/DLD-FUNCTIONS/__magick_read__.cc
+++ b/src/DLD-FUNCTIONS/__magick_read__.cc
@@ -794,16 +794,18 @@ Instead you should use @code{imwrite}.\n
 
   error ("__magick_write__: not available in this version of Octave");
 
 #endif
 
 return retval;
 }
 
+#ifdef HAVE_MAGICK
+
 template<class T>
 static octave_value
 magick_to_octave_value (const T magick)
 {
   return octave_value (magick);
 }
 
 static octave_value
@@ -868,16 +870,18 @@ magick_to_octave_value (const Magick::Im
   try \
     { \
       st.assign (OUTNAME, magick_to_octave_value (im.NAME ())); \
     } \
   catch (Magick::Warning& w) \
     { \
     }
 
+#endif
+
 DEFUN_DLD (__magick_finfo__, args, ,
   "-*- texinfo -*-\n\
 @deftypefn {Loadable File} {} __magick_finfo__(@var{fname})\n\
 Read image information with GraphicsMagick++. In general you should\n\
 not be using this function.  Instead you should use @code{imfinfo}.\n\
 @seealso{imfinfo, imread}\n\
 @end deftypefn")
 {
